python/tskit/util.py

```diff
@@ -542,6 +542,98 @@ def tree_html(tree):
+        <div>
+            <style>
+                .tskit-table thead tr th {{text-align: left;padding: 0.5em 0.5em;}}
+                .tskit-table tbody tr td {{padding: 0.5em 0.5em;}}
+                .tskit-table tbody tr td:first-of-type {{text-align: left;}}
+                .tskit-details-label {{vertical-align: top; padding-right:5px;}}
+                .tskit-table-set {{display: inline-flex;flex-wrap: wrap;margin: -12px 0 0 -12px;width: calc(100% + 12px);}}
+                .tskit-table-set-table {{margin: 12px 0 0 12px;}}
+                details {{display: inline-block;}}
+                summary {{cursor: pointer; outline: 0; display: list-item;}}
+            </style>
+            <div class="tskit-table-set">
+                <div class="tskit-table-set-table">
+                <table class="tskit-table">
+                    <thead>
+                    <tr>
+                        <th style="padding:0;line-height:21px;">
+                        <img style="height: 32px;display: inline-block;padding: 3px 5px 3px 0;" src="{url_tskit_logo}"/>
+                        <a target="_blank" href="{url_variant_class_doc}"> {class_type} </a>
+                        </th>
+                    </tr>
+                    </thead>
+                    <tbody>
+        """  # noqa: B950
+
+    html_body_tail = """
+                    </tbody>
+            </table>
+            </div>
+        </div>
+        </div>
+        """
+
+    try:
+        variant.site
+
+        site_id = variant.site.id
+        site_position = variant.site.position
+        num_samples = len(variant.samples)
+        num_alleles = variant.num_alleles
+        has_missing_data = str(variant.has_missing_data)
+        isolated_as_missing = str(bool(variant.isolated_as_missing))
+
+        counts = variant.counts()
+        freqs = variant.frequencies()
+
+        return (
+            html_body_head
+            + f"""
+                <tr><td>Site Id</td><td>{site_id}</td></tr>
+                <tr><td>Site Position</td><td>{site_position}</td></tr>
+                <tr><td>Number of Samples</td><td>{num_samples}</td></tr>
+                <tr><td>Number of Alleles</td><td>{num_alleles}</td></tr>
+            """  # noqa: B950
+            + "\n".join(
+                [
+                    f"""<tr><td>Samples with allele {'missing' if k is None
+                        else "'" + k + "'"}</td><td>"""
+                    + f"{counts[k]}"
+                    + " "
+                    + f"({freqs[k] * 100:.2g}%)"
+                    + "</td></tr>"
+                    for k in variant.alleles
+                ]
+            )
+            + f"""
+                <tr><td>Has Missing Data</td><td>{has_missing_data}</td></tr>
+                <tr><td>Isolated as Missing</td><td>{isolated_as_missing}</td></tr>
+            """  # noqa: B590
+            + html_body_tail
+        )
+    except ValueError as err:
+        return (
+            html_body_head
```

Comment:
Might as well get this covered.

Reply:
I've added this test.

```
def test_variant_html_repr_no_site(self):
        tables = tskit.TableCollection(10)
        ts = tables.tree_sequence()
        v = tskit.Variant(ts)
        html = v._repr_html_()
        ElementTree.fromstring(html)
        assert len(html) > 1600
```

But I'm not sure if 1600 is appropriate here. The length of the HTML string is 1633.